pxtblocks/blocklyloader.ts

```diff
@@ -3199,4 +3212,18 @@ namespace pxt.blocks {
             }
         }
     }
+
+    let iconCanvasCache: Map<string> = {};
+    function iconToFieldImage(id: string): Blockly.FieldImage {
+        let url = iconCanvasCache[id];
+        return new Blockly.FieldImage(url, 30, 30, false, '');
+    }
+
+    function initJresIcons(blockInfo: pxtc.BlocksInfo) {
+        Object.keys(blockInfo.apis.jres).forEach((jresId) => {
```

Comment:
Why are you not doing this lazily?

Reply:
So I don't have to pass the blockInfo all the way down the stack to the initField function.